src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts

```diff
@@ -574,10 +575,7 @@ export abstract class AbstractTaskService extends Disposable implements ITaskSer
 		// We need to first wait for extensions to be registered because we might read
 		// the `TaskDefinitionRegistry` in case `type` is `undefined`
 		await this._extensionService.whenInstalledExtensionsRegistered();
-
-		await Promise.all(
-			this._getActivationEvents(type).map(activationEvent => this._extensionService.activateByEvent(activationEvent))
-		);
+		this._getActivationEvents(type).map(activationEvent => Promise.race([this._extensionService.activateByEvent(activationEvent), timeout(2000).then(() => { return undefined; })]));
```

Comment:
You're not actually using the map here or need the `.then(() => { return undefined; })` part. This isn't actually awaiting at all anymore though, I think this is what you intended:

```ts
// Wait for the first of either to resolve
await Promise.race([
    // Create a promise that resolves when all activation promises resolve
    Promise.all(
        this._getActivationEvents(type).map(activationEvent => this._extensionService.activateByEvent(activationEvent))
    ),
    timeout(2000)
]);
```